### DISCUSSION TITLE
Is it possible to catch the text of a checkbox?

### DISCUSSION THREAD

**Discussion by sprogus** [zauberzeug/nicegui#357] [Q&A] [answered]
I would like to pass the value(true/false) and the text of a checkbox into a function but I haven't been able to figure out how to capture the text.

What I am trying to is, if checkbox is false, remove the text of the checkbox from a list. If the checkbox is true, add the text back into the list.

**Comment by rodja** [comment, marked as answer]
The ValueChangeEventArguments contain the sender and the value:

```py
from nicegui import ui, events

async def check(event: events.ValueChangeEventArguments):
    event.sender.set_text('Checked' if event.value else '')

ui.checkbox(value=False, on_change=check)

ui.run()
```

**Reply by sprogus** [reply]
Thanks. That actually sent me down the correct path. I didn't realise I could pull the text out using event.sender._text

```py
from nicegui import ui
from nicegui.events import ValueChangeEventArguments


def print_checkbox_text(event: ValueChangeEventArguments):
    if event.value:
        print(event.sender._text)


ui.checkbox("item", value=True, on_change=print_checkbox_text)


ui.run(port=5556)
```

**Reply by rodja** [reply]
You do not need to access the private field. Better to use the official `.text` property: 

```py
print(event.sender.text)
```
